tests/unit/stacked-vertical-bar-test.js

```diff
@@ -0,0 +1,773 @@
+   barLabel: "Group One",
+   value: 20
+}, {
+   sliceLabel: "Label 1",
+   barLabel: "Group Two",
+   value: 32
+}, {
+   sliceLabel: "Label 1",
+   barLabel: "Group Three",
+   value: 4
+}, {
+   sliceLabel: "Label 2",
+   barLabel: "Group One",
+   value: 16
+}, {
+   sliceLabel: "Label 2",
+   barLabel: "Group Two",
+   value: 17
+}, {
+   sliceLabel: "Label 2",
+   barLabel: "Group Three",
+   value: -18
+}, {
+   sliceLabel: "Label 3",
+   barLabel: "Group One",
+   value: -18
+}, {
+   sliceLabel: "Label 3",
+   barLabel: "Group Two",
+   value: 18
+}, {
+   sliceLabel: "Label 3",
+   barLabel: "Group Three",
+   value: -19
+}];
+
+// Adds additional slice types to the three_ranges data for higher granularity
+// and more thorough testing of slice sorting mechanics.
+var sliceSortingData = _.cloneDeep(three_ranges).concat([{
+  sliceLabel: "Label 4",
+  barLabel: "Group One",
+  value: -10
+}, {
+  sliceLabel: "Label 4",
+  barLabel: "Group Two",
+  value: -10
+}, {
+  sliceLabel: "Label 4",
+  barLabel: "Group Three",
+  value: 15
+}, {
+  sliceLabel: "Label 5",
+  barLabel: "Group One",
+  value: 10
+}, {
+  sliceLabel: "Label 5",
+  barLabel: "Group Three",
+  value: 10
+}, {
+  sliceLabel: "Label 6",
+  barLabel: "Group One",
+  value: 20
+}, {
+  sliceLabel: "Label 6",
+  barLabel: "Group Three",
+  value: 20
+}]);
+
+moduleForComponent('stacked-vertical-bar-chart', '[Unit] Stacked bar component', {
+  needs: [ 'template:components/chart-component'],
+  beforeEach: function () {},
+  afterEach: function () {}
+});
+
+test("it exists", function(assert) {
+  assert.ok(this.subject());
+});
+
+test('Margins are the right size', function(assert) {
+  assert.expect(3);
+  var component = this.subject();
+
+  assert.equal(component.get('marginLeft'), 0, 'no left margin');
+  assert.equal(component.get('marginRight'), 0, 'no right margin');
+  assert.equal(component.get('marginBottom'), 30, 'has top margin (because it has a legend)');
+});
+
+test('In total, the correct number of stacking slices is displayed', function(assert) {
+  assert.expect(1);
+  this.subject({data: three_ranges});
+  this.render();
+
+  assert.equal(this.$('svg rect').length, three_ranges.length,
+    "For " + three_ranges.length + " data points, that many stacking slices are shown");
+});
+
+// Passed in to Array.prototype.sort(), which has the bogus default behavior of
+// coercing every value in the array to a string and sorting alphabetically
+const diff = function(x, y) {
+  return (x - y);
+};
+
+const getFloatAttrValueOfSvgSlice = function(slice, attributeName) {
+  return parseFloat(slice.attributes[attributeName].value);
+};
+
+const getHeightOfSvgSlice = function(slice) {
+  return getFloatAttrValueOfSvgSlice(slice, 'height');
+};
+
+const PERCENT_TOLERANCE = 0.01;
+
+const equalsWithTolerance = function(actual, expected) {
+  var tolerance = Math.abs(expected * PERCENT_TOLERANCE);
+  return ((actual < 0.0) === (expected < 0.0)) &&
```

Comment:
If `actual` and `expected` have opposite signs (that is, `((actual < 0.0) !== (expected < 0.0))`), then I _think_ `Math.abs(actual - expected) > Math.abs(expected)` and `Math.abs(actual - expected) > Math.abs(actual)`.

Hence, you are right, as long as `PERCENT_TOLERANCE` is less than 1, we don't need to test `((actual < 0.0) === (expected < 0.0))`. But I wrote it this way because I thought it was more clear like that. I am not good at math.